app/src/utils/data/grantRounds.ts

```diff
@@ -87,120 +86,136 @@ export async function getGrantRound(blockNumber: number, grantRoundAddress: stri
   return await syncStorage(
     grantRoundKeyPrefix + grantRoundAddress,
     {
-      ts: Date.now() / 1000,
       blockNumber: blockNumber,
     },
-    async (localStorageData?: LocalStorageData | undefined, save?: () => void) => {
-      const { provider, multicall } = useWalletStore();
-      const data = localStorageData?.data || {};
-
-      // read from registry on first load and every 30 secs after (we need to pick up any changes)
-      if (
-        forceRefresh ||
-        !localStorageData ||
-        (multicall.value && localStorageData && (localStorageData.ts || 0) < Date.now() / 1000 - 30)
-      ) {
-        // open the rounds contract
-        const roundContract = new Contract(grantRoundAddress, GRANT_ROUND_ABI, provider.value);
-        // collect the donationToken before promise.all'ing everything
-        const donationTokenAddress = data?.donationToken?.address || (await roundContract.donationToken());
-        const matchingTokenAddress = data?.matchingToken?.address || (await roundContract.matchingToken());
-        // use matchingTokenContract to get balance
-        const matchingTokenContract = new Contract(matchingTokenAddress, ERC20_ABI, provider.value);
-
+    async (LocalForageData?: LocalForageData | undefined, save?: () => void) => {
+      const { provider } = useWalletStore();
+      // use the ls_blockNumber to decide if we need to update the rounds data
+      const ls_blockNumber = LocalForageData?.blockNumber || 0;
```

Comment:
We should never default to zero as the start block, since it's bad for performance. Instead let's use the `START_BLOCK` exported from `chains.ts`

Reply:
This makes sense! 🙌